presto-hive/src/main/java/com/facebook/presto/hive/metastore/CachingHiveMetastore.java

```diff
@@ -205,10 +204,11 @@ public void flushCache()
     private static <K, V> V get(LoadingCache<K, V> cache, K key)
     {
         try {
-            return cache.get(key);
+            return cache.getUnchecked(key);
         }
-        catch (ExecutionException | UncheckedExecutionException | ExecutionError e) {
-            throw Throwables.propagate(e.getCause());
+        catch (UncheckedExecutionException e) {
+            throwIfUnchecked(e.getCause());
```

Comment:
I used to do this a lot, but now i don't think it is right to unwrap.

https://github.com/google/guava/wiki/Why-we-deprecated-Throwables.propagate#encourages-unwrapping-exceptions-from-other-threads

Reply:
I've read that and I agree with it in general. However, we often have caches in Presto where the loader will throw `PrestoException` which already has the correct exception code and message.

One option is we only throw the cause if it is `PrestoException`:
```java
throwIfInstanceOf(e.getCause(), PrestoException.class);
```

Another option is to add a constructor for `PrestoException` that takes a `PrestoException` and wrappers it with the same error code and message. However, that would lose custom exception classes `TableNotFoundException` (though the message and error code would still be correct).

Reply:
> throwIfInstanceOf(e.getCause(), PrestoException.class);

Yeah, this seems to be very reasonable.